### ISSUE TITLE
Akeneo 2.3.21 - Productpage loads slow, fetches all attributes?

### ISSUE THREAD

**Issue by shopstart** [akeneo/pim-community-dev#9271]
Hello

I'm not sure if my issue qualifies as a bug. I'm not a developer, but I've learned a bit how to look into things ;) 

I am migrating my company from Akeneo 1.7 to 2.3. Most of the work is done, but I am experiencing performance issues, mostly when loading a product. We do not have these issues on 1.7, though the dataset is the same.

Let me start with our setup:

- Akeneo 2.3.21 installed on AWS machine (docker/bitnami) 
- About 3000 products
- A large amount of attributes! about 3750
- 133 categories
- 133 families
- 166 attribute groups

On a product page, typically 7 attribute groups are loaded. The total amount of attributes may vary between say 70 and 100.

Now, when loading a single product in 2.3.21 it can take a looooong time, like 10+ seconds! So I did a network inspect in Chrome

<img width="860" alt="screen shot 2018-12-18 at 10 04 38" src="https://user-images.githubusercontent.com/9823241/50143414-cab4fc80-02ac-11e9-858b-e55e70a2e1b7.png">

It looks like there are 3 requests that fetch ALL of the attribute data, not just the data for this product. 

?options%5Blimit%5D=-1
?options%5Blimit%5D=-1
?options%5Blimit%5D=-1

<img width="575" alt="screen shot 2018-12-18 at 10 04 56" src="https://user-images.githubusercontent.com/9823241/50143523-254e5880-02ad-11e9-98e9-35ec7ff4d709.png">

To me, these request look completely identical and not optimal. What can I do to improve the loading speed of our product pages?

many thanks

Paul

**Comment by Doodoune** [comment]
I close this issue,
feel free to reopen it if you still have a problem.

Regards,

**Comment by AykutCevik** [comment]
Running Akeneo in version 2.3.29. The product page is taking up to 6 seconds to load.
![image](https://user-images.githubusercontent.com/1965251/52656208-27701480-2ef6-11e9-9e8a-9b4a8e58916b.png)
Is this related to this issue maybe?

**Comment by Doodoune** [comment]
hummm... I don't think so. This issue is about having 3 times the same call to "?options%5Blimit%5D=-1".
Can you open another issue for your question?
Regards,

**Comment by shopstart** [comment]
Hello. Sorry for my tardy reply. As this bug was already fixed in 2.3.20, I would also think it must have been a cache or asset deployment issue (I was already on 2.3.21). I have been running the Bitnami Akeneo stack, and I noticed this sometimes leads to file/folder permission issues. 

In my case, the following steps helped in improving the performance a bit

- upgraded machine from T2 large to T3 large on Amazon AWS 
- reduced the amount of attributes per product (we have many)
- increased PHP max memory size 
- experimented with ElasticSearch parameters

Especially the latter gave me some trouble when updating large amount of products over the API, we applied some throttling to deal with this. 

Hope this info is some help to you!
